src/WebJobs.Script/Workers/SharedMemoryDataTransfer/ISharedMemoryManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    public interface ISharedMemoryManager
+    {
+        Task<SharedMemoryMetadata> PutBytesAsync(byte[] content);
+
+        Task<SharedMemoryMetadata> PutObjectAsync(object input);
+
+        Task<byte[]> GetBytesAsync(string mapName, long offset, long count);
+
+        bool IsSupported(object input);
+
+        void TrackSharedMemoryMapForInvocation(string invocationId, string mapName);
+
+        bool TryFreeAllResourcesForInvocation(string invocationId);
```

Comment:
nit: Rename to `TryFreeMemoryMapsForInvocation` - this follows similar naming convention as Add